python_modules/dagster/dagster/_core/definitions/freshness_checks/non_partitioned.py

```diff
@@ -0,0 +1,160 @@
+from datetime import datetime
+from typing import Optional, Sequence, Union
+
+import pendulum
+
+from dagster import _check as check
+from dagster._core.definitions.asset_check_spec import AssetCheckSeverity
+from dagster._core.definitions.metadata import (
+    MetadataValue,
+    TimestampMetadataValue,
+)
+from dagster._core.execution.context.compute import AssetExecutionContext
+from dagster._utils.schedules import is_valid_cron_string, reverse_cron_string_iterator
+
+from ..asset_check_result import AssetCheckResult
+from ..asset_checks import AssetChecksDefinition
+from ..assets import AssetsDefinition, SourceAsset
+from ..events import CoercibleToAssetKey
+from .utils import (
+    DEFAULT_FRESHNESS_SEVERITY,
+    asset_to_keys_iterable,
+    ensure_no_duplicate_assets,
+    get_last_updated_timestamp,
+    retrieve_latest_record,
+)
+
+
+def build_non_partitioned_freshness_checks(
```

Comment:
Naming thing (which I think applies to the partitioned check factory below too): in the nearish future, checks themselves will be able to be partitioned (i.e. pass / fail per partition instead of per the entire asset).  This name implies that the check will not be partitioned, not that the asset is not partitioned.